### ISSUE TITLE
Unable to add package - conflict in package

### ISSUE THREAD

**Issue by yaleman** [aptly-dev/aptly#981]
I'm trying to make a repository for a package with multiple architectures, and it's failing to add the files even though the architecture is different on each?

## Detailed Description

Trying to add the files:
```
# aptly repo add housenet debs
Loading packages...
[!] Unable to add package to repo kanidm_1.1.0-alpha.5_amd64: conflict in package kanidm_1.1.0-alpha.5_amd64
[!] Unable to add package to repo kanidm_1.1.0-alpha.5_arm64: conflict in package kanidm_1.1.0-alpha.5_arm64
[!] Some files were skipped due to errors:
  debs/ubuntu/bionic/kanidm_amd64.deb
  debs/ubuntu/bionic/kanidm_arm64.deb
ERROR: some files failed to be added
```

Files in situ (it's running in a docker container, hence root):
```
root@884e921e9b11:/data# find debs/ -ls
 75947878      0 drwxr-xr-x   3 root     root           96 Sep 24 02:01 debs/
 75947913      0 drwxr-xr-x   3 root     root           96 Sep 23 22:24 debs/ubuntu
 75947914      0 drwxr-xr-x   4 root     root          128 Sep 24 02:01 debs/ubuntu/bionic
 75947924  14712 -rw-r--r--   4 root     root     15062100 Sep 24 02:01 debs/ubuntu/bionic/kanidm_amd64.deb
 75963569  14400 -rw-r--r--   4 root     root     14712396 Sep 24 02:01 debs/ubuntu/bionic/kanidm_arm64.deb
```

amd64 package:
```
root@884e921e9b11:/data# dpkg-deb -I debs/ubuntu/bionic/kanidm_amd64.deb
 new Debian package, version 2.0.
 size 15062100 bytes: control archive=78424 bytes.
       0 bytes,     0 lines      conffiles
     187 bytes,     9 lines      control
  389076 bytes,  3934 lines      md5sums
     234 bytes,     8 lines   *  postinst             #!/bin/bash
 Package: kanidm
 Essential: no
 Section: web
 Priority: optional
 Maintainer: James Hodgkinson
 Description: Kanidm CLI Client
 Version: 1.1.0-alpha.5
 Installed-Size: 95768
 Architecture: amd64
```

arm64 package: 

```
# dpkg-deb -I debs/ubuntu/bionic/kanidm_arm64.deb
 new Debian package, version 2.0.
 size 14712396 bytes: control archive=79128 bytes.
       0 bytes,     0 lines      conffiles
     187 bytes,     9 lines      control
  389220 bytes,  3943 lines      md5sums
     234 bytes,     8 lines   *  postinst             #!/bin/bash
 Package: kanidm
 Essential: no
 Section: web
 Priority: optional
 Maintainer: James Hodgkinson
 Description: Kanidm CLI Client
 Version: 1.1.0-alpha.5
 Installed-Size: 96972
 Architecture: arm64
```

**Comment by yaleman** [comment]
never mind, I seem to have worked it out.

**Comment by sgtcoder** [comment]
> never mind, I seem to have worked it out.

What was the solution? I am having the same issues with one of my debs, Postman.

**Comment by sgtcoder** [comment]
Nevermind, have to do a `-force-replace`

**Comment by r4co0n** [comment]
@sgtcoder: You only have to use `-force-replace` if you are adding the same package, e.g. `postman`, upstream version `0.0.4`, revision `7` for architecture `amd64`, hence a Debian binary package that would be canonically named `postman_0.04-7_amd64.deb` to your repository a second time, and this second package differs from the first one.

This happens if your package build is not reproducible and executed multiple times, assuming you did not change the package while keeping the version static.

The APT clients downstream of your aptly instance are using package name, version and architecture to determine if an upgrade is necessary. You should consider a package is immutable once it has been published. Since aptly cannot know if any such relationship has been established already, it plays it safe and refuses to change packages that were previously successfully registered for this repository - The same holds true for mirrors and publications.

Long story short, I don't see why you needed to use `-force-overwrite` or `-force-replace`, unless you were changing your package and still wanting to keep it in the same repository and/or publication.

> You would think you shouldn't have to do that, of course we are replacing, we are UPDATING!!

I hope I have made the difference between replacing an already existing package version and upgrading a package, which should raise the version, clear. If not, please tell me where you still don't know why aptly does what it does.
